### ISSUE TITLE
Kodi 19.3 Freezes With HTTPS Traffic

### ISSUE THREAD

**Issue by jwor80** [xbmc/xbmc#20631]
### Describe the bug
Here is a clear and concise description of what the problem is:

I have a laptop setup with Kodi 19.3 using the Jellyfin & Emby-next-gen addons (seperate profiles). When I configure the URL for these addons to use HTTPS I see issues with Kodi freezing and becoming unresponsive when playing a video or when stopping a video.  The issue does not occur when I use an HTTP address.  I did not see this issue with Kodi 18.9.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The video should play and stop without kodi freezing and stop responding.

## Actual Behavior

Kodi will freeze and become unresponsive when I play a video.  If the video does play then it will more than likely freeze and become unresponsive when I attempt to stop the video.

### To Reproduce

1. Configure the emby-next-gen or jelly addon with an HTTPS connection
2. Play and Stop a video from the library

### Debuglog

The debuglog can be found here:
[kodi.log](https://github.com/xbmc/xbmc/files/7655123/kodi.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

![Capture](https://user-images.githubusercontent.com/38135239/144731724-4b74895a-633e-4e1e-ae9a-194adcdee11c.JPG)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [X ] Windows 10
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 19.3

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ltGuillaume** [comment]
The same thing happens with a regular HTTPS nginx server and a bookmark in Kodi to a directory listing: on Kodi 18.9, no issues, on Kodi 19.x, playback is impossible due to freezes of both video and UI. Larger buffers via advancedsettings.xml did not resolve this.

**Comment by sopparus** [comment]
Can you guys try with disabled http2?
https://kodi.wiki/view/Advancedsettings.xml#network

**Comment by jwor80** [comment]
The disabling of https2 in the has resolved my issue.

**Comment by wsnipex** [comment]
please provide a full debug log with curl component logging when using nginx.

**Comment by ltGuillaume** [comment]
That seems to work for me, too.

@wsnipex Here's a debug+libCURL log to a Jellyfin instance using nginx. I started the video, which started playback after ~7 seconds for only a few frames, then Kodi froze. After 2 minutes I pressed the hotkey 'X' to close the video, after which Windows immediately showed "Kodi (not responding)" in the title bar. After 2 minutes, I pressed the window's X button, then "End process" in the resulting dialog.
[kodi19.3-http2curlproblem.zip](https://github.com/xbmc/xbmc/files/7663769/kodi19.3-http2curlproblem.zip)

**Comment by sopparus** [comment]
same issue with master, just tested

**Comment by ltGuillaume** [comment]
> edit: this is only a problem on windows

Is it? I'm getting the exact same _behavior_ at least on a Philips Android TV...

> Note that we don't support jellyfin, so I explicitly asked for a log without it

[kodi19.3-nginxhttp2curlproblem.zip](https://github.com/xbmc/xbmc/files/7678936/kodi19.3-nginxhttp2curlproblem.zip)

**Comment by ltGuillaume** [comment]
> Note that we don't support jellyfin, so I explicitly asked for a log without it
@wsnipex 
[kodi19.3-nginxhttp2curlproblem.zip](https://github.com/xbmc/xbmc/files/7678999/kodi19.3-nginxhttp2curlproblem.zip)

**Comment by wsnipex** [comment]
@ltGuillaume I don't see any errors besides a lot of those:

> 2021-12-08 21:55:04.778 T:4300     INFO <general>: JELLYFIN.service -> ERROR::service.py:182 Exception in Playback Monitor: Expecting value: line 1 column 1 (char 0)
2021-12-08 21:55:04.778 T:4300     INFO <general>: JELLYFIN.service -> ERROR::service.py:183 Traceback (most recent call last):
                                                     File "E:\Kodi19\portable_data\addons\plugin.video.jellycon\service.py", line 127, in <module>
                                                       send_progress()
                                                     File "E:\Kodi19\portable_data\addons\plugin.video.jellycon\resources\lib\play_utils.py", line 876, in send_progress
                                                       play_data = json.loads(play_data_string)
                                                     File "E:\Kodi19\system\python\Lib\json\__init__.py", line 357, in loads
                                                       return _default_decoder.decode(s)
                                                     File "E:\Kodi19\system\python\Lib\json\decoder.py", line 337, in decode
                                                       obj, end = self.raw_decode(s, idx=_w(s, 0).end())
                                                     File "E:\Kodi19\system\python\Lib\json\decoder.py", line 355, in raw_decode
                                                       raise JSONDecodeError("Expecting value", s, err.value) from None
                                                   json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

doesn't look like a log without jellyfin. And it isn't unmodified either.

**Comment by ltGuillaume** [comment]
The only thing I modified are the filenames.
The only thing Jellyfin-related is because the plugin is awfully verbose.
This was a playback via simple HTTPS directory listing on an nginx server, nothing via Jellyfin.
It was even worse than via Jellyfin, because Kodi just showed me a spinner for 2-3 minutes, not even starting playback of a single frame, and crashed subsequently when I tried to press X. (After disabling HTTP2 it ran perfectly fine.)
